test/memory.t.cpp

```diff
@@ -0,0 +1,72 @@
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include <iostream>
+
+#include "memory.hpp"
+#include "iromloader.hpp"
+#include "ifileio.hpp"
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::SetArgPointee;
+using ::testing::SetArrayArgument;
+
+class MockFileIo : public IFileIO
+{
+public:
+    MOCK_METHOD1(openFile, int(std::string const &));
+    MOCK_METHOD2(readFile, int(uint8_t*, int));
+    MOCK_METHOD1(closeFile, void(int));
+};
+
+class MockRomLoader : public IRomLoader
+{
+public:
+    MOCK_METHOD1(load, bool(std::string const&));
+    MOCK_METHOD0(getData, std::vector<uint8_t>());
+};
+
+class MemoryTest : public ::testing::Test
+{
+public:
+    MemoryTest();
+    MockFileIo _fileIO;
+    MockRomLoader _romLoader;
+    std::array<uint8_t, 0x200000> _cartridge;
+    std::array<uint8_t, 0x200000> _emptyCartridge;
+    std::array<uint8_t, 0x4000> _bank0;
+private:
+};
+
+MemoryTest::MemoryTest()
```

Comment:
move this function inside the class it self, it's contain the Fixture so that when we are reading we don't get any noise